### PR TITLE
Added support for Adobe Client Data Layer

### LINKED ISSUE #447: Add support for Adobe Client Data Layer
With [Core Components version 2.9.0](https://github.com/adobe/aem-core-wcm-components/releases/tag/core.wcm.components.reactor-2.9.0), support was added for [Adobe Client Data Layer](https://github.com/adobe/adobe-client-data-layer). The archetype should allow adding this kind of support to customer projects.

### PR DESCRIPTION
**Pull request by vladbailescu**
Fixes #447

### CODE REVIEW

**Comment by vladbailescu** [comment]
Please not, when testing, to use new `groupId` (#406):

```
mvn -B archetype:generate \                                                                                                                           1.8m  Fri Aug  7 16:08:50 2020
                 -D archetypeGroupId=com.adobe.aem \
                 -D archetypeArtifactId=aem-project-archetype \
                 -D archetypeVersion=24-SNAPSHOT \
                 -D aemVersion=cloud \
                 -D appTitle="My Site" \
                 -D appId="mysite" \
                 -D groupId="com.mysite" \
                 -D frontendModule=general \
                 -D includeExamples=n \
                 -D datalayer=y
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__appId__/.content.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
         cq:conf="/conf/${appId}"
         jcr:primaryType="cq:PageContent"
         jcr:title="${appTitle}"
+        sling:configRef="/conf/${appId}"
```
Comment (bpauli) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__appId__/.content.xml:9]: image

anonymous access on publish is not possible for the `sling:conf` folder. Please add ACL entry with `rep:glob` on `/conf` accordingly.

Comment (vladbailescu) [reply src/main/archetype/ui.content/src/main/content/jcr_root/content/__appId__/.content.xml:9]: I already added access to `/conf/_appId_/_sling_configs` in https://github.com/adobe/aem-project-archetype/pull/448/files#diff-5f93dfb9442aa73184a52dae218f20a1

Comment (bpauli) [reply src/main/archetype/ui.content/src/main/content/jcr_root/content/__appId__/.content.xml:9]: Sorry for the noise. The issue was just an outdated core bundle.

File: src/test/resources/projects/basic/archetype.properties [hunk 2]
```diff
@@ -14,3 +14,4 @@ includeErrorHandler=n
 includeDispatcherConfig=n
 includeCommerce=n
 commerceEndpoint=https://hostname.com/grapql
+datalayer=y
```
Comment (jckautzmann) [inline src/test/resources/projects/basic/archetype.properties:17]: shouldn't it be `datalayer=${datalayer}` as for the below projects?

Comment (vladbailescu) [reply src/test/resources/projects/basic/archetype.properties:17]: `datalayer=${datalayer}` uses the default value. For `basic` I intended to use have it on, regardless.

Comment (jckautzmann) [reply src/test/resources/projects/basic/archetype.properties:17]: ok, makes sense.

**Pull request merged**